avocado/core/task/statemachine.py

```diff
@@ -247,6 +247,7 @@ async def start(self):
             await asyncio.sleep(0.1)
             return
 
+        await self._spawner.create_task_output_dir(runtime_task)
```

Comment:
Personally, I think the call to create the output-dir should not be the responsibility of the state machine.